app/src/androidTest/java/com/android/sample/ui/workout/WorkoutScreenTest.kt

```diff
@@ -148,19 +194,124 @@ class WorkoutScreenTest {
+    composeTestRule.onNodeWithTag("recordSwitch").assertIsDisplayed()
+
+    // VideoPlayer
+    composeTestRule.onNodeWithTag("VideoPlayer").assertIsDisplayed()
+
     // ClickOnStart to start the 1st activity
     composeTestRule.onNodeWithTag("StartButton").performClick()
+  }
+
+  @Test
+  fun presentationIsDisplayedWarmup() {
+    composeTestRule.setContent {
+      WorkoutScreen(
+          navigationActions,
+          bodyweightViewModel = bodyWeightViewModel,
+          yogaViewModel = yogaViewModel,
+          warmUpViewModel = warmUpViewModel,
+          workoutType = WorkoutType.YOGA,
+          videoViewModel = mockVideoViewModel)
+    }
+    // ArrowBack
+    composeTestRule.onNodeWithTag("ArrowBackButton").assertIsDisplayed()
 
+    // Test if the presentation screen is well displayed
+    // Workout Name
+    composeTestRule.onNodeWithTag("WorkoutName").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("WorkoutName").assertTextEquals("MyWorkout")
+    // Exercise name
+    composeTestRule.onNodeWithTag("ExerciseName").assertIsDisplayed()
+    composeTestRule
+        .onNodeWithTag("ExerciseName")
+        .assertTextEquals(
+            bodyWeightViewModel.selectedWorkout.value?.exercises?.get(0)?.type.toString())
+    // Exercise description
+    composeTestRule.onNodeWithTag("ExerciseDescription").assertIsDisplayed()
+    bodyWeightViewModel.selectedWorkout.value?.exercises?.get(0)?.type?.let {
+      composeTestRule.onNodeWithTag("ExerciseDescription").assertTextEquals(it.getInstruction())
+    }
+
+    // Goal Icon and value
+
+    composeTestRule.onNodeWithTag("GoalIcon").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("GoalValue").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("GoalValue").assertTextEquals("3 Rep.")
+
+    // Skip and start button are displayed
+
+    composeTestRule.onNodeWithTag("SkipButton").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("SkipButton").assertTextEquals("Skip")
+
+    composeTestRule.onNodeWithTag("StartButton").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("StartButton").assertTextEquals("Start")
+    composeTestRule.onNodeWithTag("recordSwitch").assertIsDisplayed()
+
+    // VideoPlayer
+    composeTestRule.onNodeWithTag("VideoPlayer").assertIsDisplayed()
+
+    // ClickOnStart to start the 1st activity
+    composeTestRule.onNodeWithTag("StartButton").performClick()
+  }
+
+  @Test
+  fun startingExerciseIsDisplayed() {
+    composeTestRule.setContent {
+      WorkoutScreen(
+          navigationActions,
+          bodyweightViewModel = bodyWeightViewModel,
+          yogaViewModel = yogaViewModel,
+          warmUpViewModel = warmUpViewModel,
+          workoutType = WorkoutType.BODY_WEIGHT,
+          videoViewModel = mockVideoViewModel)
+    }
+
+    // ClickOnStart to start the 1st activity
+    composeTestRule.onNodeWithTag("StartButton").performClick()
+    verify(mockVideoRepository, atLeast(1)).getVideos(any(), any())
```

Comment:
Nice test !

But the title doesn't say that you verify this.

Reply:
This actually starts the 1s exercise of the workout, maybe I didnt understand your comment ?